### PR TITLE
Automate issue assignment and implement inactivity management

### LINKED ISSUE #6647: [Bug]: Stale task reminder and auto-unassign after inactivity
### Summary

Enhancement to the bot to handle inactivity on assigned issues by sending gentle reminders and unassigning inactive assignees after a grace period, to keep work flowing.

Plan : After `X` days of inactivity on an issue with an assignee, send a gentle reminder to the assignee (e.g., “Hey @ user, just checking in!...........”).
If there is still no response after ~2 more days, optionally unassign the current assignee to open the issue for others.
Keep the issue open (no automatic locking) unless/until the unassignment happens.

### Steps to reproduce

NA

### Expected behaviour

After `X` days of inactivity on an assigned issue, a gentle reminder is posted to the assignee.
If there is no response after `~2` additional days, the bot unassigns the issue to open it up for others, while keeping the issue accessible.

### Actual behaviour

Currently, there is no automated reminder or unassignment mechanism for inactive issues.

### Device name

N/A

### Android version

N/A

### Commons app version

6.2

### Device logs

_No response_

### Screenshots

_No response_

### Would you like to work on the issue?

Edit : No

### PR DESCRIPTION
**Pull request by Kota-Jagadeesh**
**Description (required)**

Fixes #6647

What changes did you make and why?
Introduced a major improvements to our workflow automation:

1. **Inactivity Manager (`inactivity_manager.yml`):**
   * **3-Stage Lifecycle:** Implemented the a proactive 30/44/58-day schedule that sends gentle reminders before unassigning the inactive users.
   * Added a feature where the GitHub Actions bot checks open issues and sends a reminder if the assignee has not shown progress for 30 days. If there is still no progress after the reminder, and the assignee remains inactive for 2 weeks, the bot sends a final warning message. If there is still no response, the bot unassigns them. This workflow was placed in a new file because mixing it with auto_assign.yml didn’t feel right. To set a softer tone, added the emojis. For now, I’ll keep it as a draft and monitor it for a few days to spot any issues or edge cases.
   
**Tests performed (required)**
N/A

**Screenshots (for UI changes only)**
N/A (Workflow/Backend changes).

### CODE REVIEW

File: .github/workflows/inactivity_manager.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        id: ai-brain
+        with:
+          model: openai/gpt-4o-mini
+          prompt: |
+            You are a project manager. Below is a list of stale GitHub issues in JSON format.
+            For each issue, analyze the "lastComment" field.
+            
+            Determine if the assignee has provided a valid reason for a temporary delay.
+            Valid reasons: Exams, health issues, bereavement, or specific "busy until [date]" notes.
+            Invalid reasons: Silence, "I'm working on it" with no progress for 30 days, or general "I'm busy."
```
Comment (RitikaPahwa4444) [inline .github/workflows/inactivity_manager.yml]: I know this is a draft but let's not define invalid reasons except no response for quite some time. An LLM could be insensitive to understand the situation. "I'm working on it" or "I'm busy" are equally valid reasons - not everyone is comfortable sharing their reasons for inactivity.

**Comment by Kota-Jagadeesh** [comment]
Updates made:

Moved from using the AI to the workflow by using a keyword-matching system. now, If the bot detects that an issue is assigned but no work is happening, and there has been no progress for 30 days, the bot sends a gentle reminder to the assignee. and ,if the assignee does not respond even after the reminder and it waits 14 more days for a reply, and then the bot sends a final reminder. if there is still no reply after the final reminder (here it waits for the reply from the assignee for 14 days after the final remainder is sent), the issue is unassigned and the issue is kept open for other users to take on.

**Edit** 

Case 2:

If a user is assigned but needs more time due to personal reasons, the bot will leave a tip in the comments suggesting them reply with the hashtag **#keep**. If the user includes **#keep** in their reply, the bot recognizes this as an active request to stay on the task. It then pauses any unassignment actions and resets the timer, giving the original assignee another 30 days of protected time.

File: .github/workflows/inactivity_manager.yml (hunk 1 above)
Comment (Kota-Jagadeesh) [reply .github/workflows/inactivity_manager.yml]: hi @RitikaPahwa4444, I’ve implemented new changes that do not rely on thee LLM. please review them and let me know if any further adjustments are needed 🙂

Comment (RitikaPahwa4444) [reply .github/workflows/inactivity_manager.yml]: Hey @Kota-Jagadeesh I'm on-call at work this week, will try to review your PRs by the weekend if I find some time.

Comment (Kota-Jagadeesh) [reply .github/workflows/inactivity_manager.yml]: No problem @RitikaPahwa4444!

**Comment by Kota-Jagadeesh** [comment]
@RitikaPahwa4444 should i close this PR or keep it as a draft? since we're now not using the github actions workflow for auto-assignment

**Comment by RitikaPahwa4444** [comment]
Sorry @Kota-Jagadeesh for taking a different route while you worked on this; had to move to the old triage and assign to avoid features we don't intend to build at all.

I'm okay with either as both can serve as an inspiration in case we decide to automate it again :)

**Comment by RitikaPahwa4444** [comment]
I added the label I use to track such features. We can close for now.

**Comment by Kota-Jagadeesh** [comment]
it's  fine @RitikaPahwa4444, for now i'm closing this
